src/main/java/com/code_intelligence/jazzer/mutation/support/InputStreamSupport.java

```diff
@@ -193,5 +193,67 @@ public void close() throws IOException {
     }
   }
 
+  /**
+   * Wraps a given stream with the functionality to detect if it was consumed exactly.
+   * To do so, the stream must provide an accurate implementation of {@link
```

Comment:
Should we enforce this by limiting it to accepting `ByteArrayInputStream`s? For input streams doing actual IO we can't really guarantee this property.

Reply:
I didn't want to change the signature of `ArgumentsMutator#read`, which would be necessary to do that. Come to think of it, `ArgumentsMutator` hides the restriction and could provide an incorrect return value. We could pull out the `extendWithReadExactly` to the call sites. WDYT?

Reply:
We could also restrict the signature of `read` to `ByteArrayInputStream` for now and consider alternatives when we want to pass in other types of streams.

Reply:
It's not as nice as accepting `InputStream` but for sure the easier solution for now. Done.